master/internal/configpolicy/task_config_policy_intg_test.go

```diff
@@ -0,0 +1,84 @@
+package configpolicy
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+func TestPriorityAllowed(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	// When no constraints are present, any priority is allowed.
+	ok, err := PriorityAllowed(1, model.NTSCType, 0, true)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	wkspLimit := 50
+	w := addWorkspacePriorityLimit(t, pgDB, wkspLimit)
+
+	// Priority is outside workspace limit.
+	ok, err = PriorityAllowed(w.ID, model.NTSCType, wkspLimit-1, true)
+	require.NoError(t, err)
+	require.False(t, ok)
+
+	globalLimit := 42
+	addGlobalPriorityLimit(t, pgDB, globalLimit)
+
+	// Priority is within global limit.
+	ok, err = PriorityAllowed(w.ID, model.NTSCType, wkspLimit-1, true)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	// Priority is outside global limit.
+	ok, err = PriorityAllowed(w.ID+1, model.NTSCType, globalLimit-1, true)
```

Comment:
```suggestion
	ok, err = PriorityAllowed(w.ID, model.NTSCType, globalLimit-1, true)
```

Reply:
This was by intent; I wanted to ensure that no workspace-level task config policies were set and so the workspace ID is one that does not exist or has no policies.

Reply:
Ooh gotcha! This makes sense